src/script/components/MessagesList/index.tsx

```diff
@@ -59,6 +60,29 @@ interface MessagesListParams {
   showUserDetails: (user: User) => void;
 }
 
+const filterSpecificDuplicatedMemberMessages = (messages: MessageEntity[]) => {
+  const typesToFilter = ['conversation.member-join', 'conversation.group-creation'];
+  return messages.reduce<MessageEntity[]>((acc, message) => {
+    if (isMemberMessage(message)) {
+      const existingMemberMessages = acc.filter(isMemberMessage);
+      if (typesToFilter.includes(message.type) && existingMemberMessages.length) {
+        switch (message.type) {
+          case 'conversation.group-creation':
+            return acc;
+          case 'conversation.member-join':
+            return existingMemberMessages.some(m => m.htmlCaption() === message.htmlCaption())
```

Comment:
dont we have a lint rule against short vars?

Reply:
Doesn't seem like :D. To be fair this is code that github copilot generated.
could also call it existingMemberMessage but since its a fairly short statement I thought m is fine